src/RefractDataStructure.cc

```diff
@@ -446,55 +405,97 @@ namespace drafter {
                     types.push_back(f.Create(it->symbol.literal, method));
                 }
 
-                return std::make_tuple(types, *NodeInfo<typename T::ValueType>::NullSourceMap(), true);
+                return std::make_tuple(types, *NodeInfo<typename T::ValueType>::NullSourceMap());
             }
         };
 
-        ExtractTypeDefinition(ElementData<ElementType>& data, ConversionContext& context) : data(data), context(context) {}
 
-        void operator()(const NodeInfo<mson::TypeDefinition>& typeDefinition) {
-            ValueInfo value = Fetch<typename T::ValueType>()(typeDefinition.node->typeSpecification.nestedTypes, context);
+        template <typename E, typename IsPrimitive = typename ElementData<E>::IsPrimitive, bool dummy = true> struct Store;
 
-            if (std::get<2>(value)) {
-                data.values.push_back(value);
+        template <typename E, bool dummy> 
+        struct Store<E, PrimitiveType, dummy> {
+            void operator()(ElementData<E>& data, const mson::TypeNames& typeNames, const ConversionContext& context){
+                // do nothing
```

Comment:
This comment doesn't add any value. It is stating what this does not why. The what it does it clear from the code (it literally does nothing), the why is unclear so might be worth adding why (or removing the comment).